lib/nsq/connection.rb

```diff
@@ -55,7 +55,7 @@ def initialize(opts = {})
       end
 
       # for outgoing communication
-      @write_queue = Queue.new
+      @write_queue = SizedQueue.new(10000)
```

Comment:
@chen-anders Do you think this is a reasonable default (see context here: https://github.com/wistia/nsq-ruby/issues/41). Figured you'd have a good sense based on our internal operations and how bursty stuff can be.

Reply:
This seems reasonable though I wonder if there's a benefit to have this being configurable.

Reply:
@chen-anders and I just caught up about this. Feels fine to start with this as the default and maybe make it configurable if there's a case for it down the road. 👍